src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+        KC_METRO,
+        KITSAP_TRANSIT,
+        PIERCE_COUNTY_TRANSIT,
+        SEATTLE_STREET_CAR,
+        SOUND_TRANSIT,
+        WASHINGTON_STATE_FERRIES
+    }
+
+    // If set to true, the test ride price is used instead of the actual agency cash fare.
+    public boolean IS_TEST;
+    public static final float DEFAULT_TEST_RIDE_PRICE = 3.49f;
+
+    public OrcaFareServiceImpl(Collection<FareRuleSet> regularFareRules) {
+        addFareRules(Fare.FareType.regular, regularFareRules);
+        addFareRules(Fare.FareType.senior, regularFareRules);
+        addFareRules(Fare.FareType.youth, regularFareRules);
+        addFareRules(Fare.FareType.orcaRegular, regularFareRules);
+        addFareRules(Fare.FareType.orcaYouth, regularFareRules);
+        addFareRules(Fare.FareType.orcaLift, regularFareRules);
+        addFareRules(Fare.FareType.orcaSenior, regularFareRules);
+    }
+
+    /**
+     * Classify the ride type based on the route information provided. In most cases the agency name is sufficient. In
+     * some cases the route description and short name are needed to define inner agency ride types.
+     */
+    private static RideType classify(Route routeData) {
+        if ("29".equals(routeData.getAgency().getId())) {
+            try {
+                int routeId = Integer.parseInt(routeData.getShortName());
+                if (routeId >= 400 && routeId <= 899) {
+                    return RideType.COMM_TRANS_COMMUTER_EXPRESS;
+                }
+                return RideType.COMM_TRANS_LOCAL_SWIFT;
+            } catch (NumberFormatException e) {
+                LOG.warn("Unable to determine comm trans route id from {}.", routeData.getShortName(), e);
+                return RideType.COMM_TRANS_LOCAL_SWIFT;
+            }
+        } else if ("1".equals(routeData.getAgency().getId())) {
+            if (routeData.getType() == 4 && routeData.getDesc().contains("Water Taxi: West Seattle")) {
+                return RideType.KC_WATER_TAXI_WEST_SEATTLE;
+            } else if (routeData.getType() == 4 && routeData.getDesc().contains("Water Taxi: Vashon Island")) {
+                return RideType.KC_WATER_TAXI_VASHON_ISLAND;
+            }
+            return RideType.KC_METRO;
+        } else if ("40".equals(routeData.getAgency().getId())) {
+            return RideType.SOUND_TRANSIT;
+        } else if ("97".equals(routeData.getAgency().getId())) {
+            return RideType.EVERETT_TRANSIT;
+        } else if ("3".equals(routeData.getAgency().getId())) {
+            return RideType.PIERCE_COUNTY_TRANSIT;
+        } else if ("23".equals(routeData.getAgency().getId())) {
+            return RideType.SEATTLE_STREET_CAR;
+        } else if ("wsf".equalsIgnoreCase(routeData.getAgency().getId())) {
+            return RideType.WASHINGTON_STATE_FERRIES;
+        } else if ("kt".equalsIgnoreCase(routeData.getAgency().getId())) {
+            return RideType.KITSAP_TRANSIT;
+        }
+        return null;
```

Comment:
All these if / else if statements could be avoided by doing something like this:

```java
// declaration of field within OcraFareServiceImpl
Map<String, Function<Route, RideType>> classificationStrategy = new HashMap<>();
classificationStrategy.put("40", rd -> RideType.SOUND_TRANSIT);

// this method:
Function<Route, RideType> classifier = classificationStrategy.get(routeData.getAgency().getId());
return classifier != null ? classifier.apply(routeData) : null;
```

Reply:
And better yet, there could be subclasses for each of these agencies that have the methods `getYouthFare`, `getLiftFare`, `getSeniorFare`, and `getRegularFare`. These can result in better organization of logic that is associated with a particular transit agency and its behavior for generating a fare.

Recently I read an article about [refactoring switch statements](https://refactoring.guru/smells/switch-statements) that says:

> As a rule of thumb, when you see `switch` you should think of polymorphism.

Reply:
Went with the function approach as this is unlikely to change much, if at all. Plus all the cost types (e.g. getLiftFare) are under a single method.